src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4; // 16
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private Node<K, V> nullKeyNode;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (nullKeyNode == null) {
+                nullKeyNode = new Node<>(0, null, value, null);
+                size++;
+                return;
+            } else {
+                nullKeyNode.value = value;
+                return;
+            }
+        }
 
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (node.key == key || node.key.equals(key))) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return nullKeyNode == null ? null : nullKeyNode.value;
+        }
+
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (node.key == key || node.key.equals(key))) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int indexFor(int hash, int size) {
+        return Math.abs(hash % size);
+    }
+
+    private int hash(K key) {
```

Comment:
The 'hash' method has a utility purpose and is only used inside this class, therefore it should not be public. Consider changing its access modifier to 'private'.